pkg/services/publicdashboards/api/api.go

```diff
@@ -164,19 +164,19 @@ func (api *Api) UpdatePublicDashboard(c *contextmodel.ReqContext) response.Respo
 		return response.Err(ErrInvalidUid.Errorf("UpdatePublicDashboard: invalid Uid %s", uid))
 	}
 
-	pd := &PublicDashboard{}
-	if err := web.Bind(c.Req, pd); err != nil {
+	pdDTO := &PublicDashboardDTO{}
+	if err := web.Bind(c.Req, pdDTO); err != nil {
 		return response.Err(ErrBadRequest.Errorf("UpdatePublicDashboard: bad request data %v", err))
 	}
 
 	// Always set the orgID and userID from the session
-	pd.OrgId = c.OrgID
-	pd.Uid = uid
+	pdDTO.OrgId = c.OrgID
+	pdDTO.Uid = uid
 	dto := SavePublicDashboardDTO{
 		UserId:          c.UserID,
 		OrgId:           c.OrgID,
```

Comment:
This always confuses me, why do we have two `OrgIds`? Maybe we could leverage this change and remove one of those or maybe both, and use the orgId of the user whenever we need it.

Reply:
Also, as a general rule, I prefer to not mutate the DTO. If we need to use the `uid`, instead of changing the received `dto` we can assign it directly to the entity we want to persist in the service.

Reply:
yes, I agree in both. I can make those changes